### PR TITLE
Introduce GitHub Action to apply team labels based on reviewers

### PR DESCRIPTION
**Pull request by mads-hartmann**
## Description

As part of moving to CODEOWNERS we will be removing the OWNERS files and thus break the existing method of applying team labels (see [RFC](https://www.notion.so/gitpod/Using-GitHub-for-PR-review-automation-35d5e77a7e4b4999a72b5eef2458f636) for details on current and the upcoming process and their implementation).

This PR introduces a GitHub Action which will apply team labels based on who is tagged as reviewers. As CODEOWNERS will be mapping paths to reviewers, this means that we effectively have a way to apply team labels based on who owns the files that were modified.

This has been used in the ops repository for a while now.

## Related Issue(s)

Part of https://github.com/gitpod-io/ops/issues/840

## How to test

Testing changes to GitHub Actions can be a bit tedious so I have been using a test repository to experiment with this. See this [PR](https://github.com/gitpod-io/codeowners-test/pull/11) that shows the appropriate labels are applied. 

Once merged I will keep an eye on PRs in this repo to make sure the right labels are indeed applied.

## Release Notes

```release-note
NONE
```

## Documentation

N/A

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **utam0k** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @utam0k` in a comment when ready.

Associated issue: *#840*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by ArthurSens — APPROVED** [review]
I can see that this very PR is a working example :)

The action indeed added the correct label

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 9512cb0566d5648ec18b02c0ed0741114e0d9915</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7977#pullrequestreview-870748922" title="Approved">ArthurSens</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **utam0k** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @utam0k` in a comment when ready.

Associated issue: *#840*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
